src/legacy/ui/public/vis/editors/default/components/default_editor_agg.tsx

```diff
@@ -0,0 +1,287 @@
+  let disabledParams;
+  let aggError;
+  const isLastBucketAgg =
+    groupName === 'buckets' && lastParentPipelineAggTitle && isLastBucket && agg.type;
+
+  const SchemaComponent = agg.schema.editorComponent;
+
+  if (isLastBucketAgg) {
+    if (['date_histogram', 'histogram'].includes(agg.type.name)) {
+      disabledParams = ['min_doc_count'];
+    } else {
+      aggError = i18n.translate('common.ui.aggTypes.metrics.wrongLastBucketTypeErrorMessage', {
+        defaultMessage:
+          'Last bucket aggregation must be "Date Histogram" or "Histogram" when using "{type}" metric aggregation.',
+        values: { type: lastParentPipelineAggTitle },
+        description: 'Date Histogram and Histogram should not be translated',
+      });
+    }
+  }
+
+  useEffect(() => {
+    if (isLastBucketAgg && ['date_histogram', 'histogram'].includes(agg.type.name)) {
+      onAggParamsChange(
+        agg.params,
+        'min_doc_count',
+        // "histogram" agg has an editor for "min_doc_count" param, which accepts boolean
+        // "date_histogram" agg doesn't have an editor for "min_doc_count" param, it should be set as a numeric value
+        agg.type.name === 'histogram' ? true : 0
+      );
+    }
+  }, [lastParentPipelineAggTitle, isLastBucket, agg.type]);
+
+  // Returns a description of the aggregation, for display in the collapsed agg header
+  const getDescription = () => {
+    if (!agg.type || !agg.type.makeLabel) {
+      return '';
+    }
+    return agg.type.makeLabel(agg) || '';
+  };
+
+  const onToggle = (isOpen: boolean) => {
+    setIsEditorOpen(isOpen);
+    if (!isOpen) {
+      setTouched(true);
+    }
+  };
+
+  const onSetValidity = (isValid: boolean) => {
+    setValidity(isValid);
+    setValidState(isValid);
+  };
+
+  const renderAggButtons = () => {
+    const actionIcons = [];
+
+    if (agg.enabled && isRemovable) {
+      actionIcons.push({
+        id: 'disableAggregation',
+        color: 'text',
+        type: 'eye',
+        onClick: () => onToggleEnableAgg(agg, false),
+        ariaLabel: i18n.translate('common.ui.vis.editors.agg.disableAggButtonAriaLabel', {
+          defaultMessage: 'Disable aggregation',
+        }),
+        tooltip: i18n.translate('common.ui.vis.editors.agg.disableAggButtonTooltip', {
+          defaultMessage: 'Disable aggregation',
+        }),
+        dataTestSubj: 'disableAggregationBtn',
+      });
+    }
+    if (!agg.enabled) {
+      actionIcons.push({
+        id: 'enableAggregation',
+        color: 'text',
+        type: 'eyeClosed',
+        onClick: () => onToggleEnableAgg(agg, true),
+        ariaLabel: i18n.translate('common.ui.vis.editors.agg.enableAggButtonAriaLabel', {
+          defaultMessage: 'Enable aggregation',
+        }),
+        tooltip: i18n.translate('common.ui.vis.editors.agg.enableAggButtonTooltip', {
+          defaultMessage: 'Enable aggregation',
+        }),
+        dataTestSubj: 'disableAggregationBtn',
```

Comment:
The action is `enable aggregation`, but data-test-subj is `disable ...` ? Does it make sense to rewrite it to `toggle...` since they both use the same `data-test-subj` attribute ?